src/algorithms/expected_consensus/_index.md

```diff
@@ -190,15 +190,26 @@ EC enforces a version of soft finality whereby all miners at round N will reject
 
 Due to the existence of potential forks in EC, a miner can try to unduly influence protocol fairness. This means they may choose to disregard the protocol in order to gain an advantage over the power they should normally get from their storage on the network. A miner should be slashed if they are provably deviating from the honest protocol.
 
-This is detectable when a given miner submits two blocks that satisfy any of the following "consensus faults":
+This is detectable when a given miner submits two blocks that satisfy any of the following "consensus faults". In all cases, we must have:
 
-- (1) `double-fork mining fault`: two blocks mined at the same epoch.
+- both blocks were mined by the same miner
+- both blocks have valid signatures
+- first block's epoch is smaller or equal than second block
+
+Thereafter, the faults are:
+
+- (1) `double-fork mining fault`: two blocks mined at the same epoch. 
+  - `blockA.Epoch == blockB.Epoch`
 {{< diagram src="diagrams/double_fork.dot.svg" title="Double-Fork Mining Fault" >}}
 
 - (2) `time-offset mining fault`: two blocks mined off of the same Tipset at different epochs (i.e. with different `ChallengeTickets` generated from the same input ticket).
+  - `blockA.Parents == blockB.Parents && blockA.Epoch != blockB.Epoch`
 {{< diagram src="diagrams/time_offset.dot.svg" title="Time-Offset Mining Fault" >}}
 
-- (3) `parent-grinding fault`: one block's parent is a Tipset that provably should have included a given block but does not. While it cannot be proven that a missing block was willfully omitted in general (i.e. network latency could simply mean the miner did not receive a particular block), it can when a miner has successfully mined a block two epochs in a row and omitted one. That is, this condition should be evoked when a miner omits their own prior block. When a miner's block at epoch e + 1 references a Tipset that does not include the block they mined at e both blocks can be submitted to prove this fault.
+- (3) `parent-grinding fault`: one block's parent is a Tipset that provably should have included a given block but does not. While it cannot be proven that a missing block was willfully omitted in general (i.e. network latency could simply mean the miner did not receive a particular block), it can when a miner has successfully mined a block two epochs in a row and omitted one. That is, this condition should be evoked when a miner omits their own prior block.
+  - Specifically, this can be proven with a "witness" block, that is by submitting blocks A, B, C where C is B's parent and A's sibling but A is not B's parent.
+  - `!blockB.Parents.Include(blockA) && blockB.Parents.Include(blockC) && blockA.Parents == blockC.Parents && blockA.Height == blockC.Height`
```

Comment:
`blockA.Height == blockC.Height` is a double-fork mining fault already

Reply:
Detailed in https://github.com/filecoin-project/go-filecoin/pull/3983#discussion_r407541095, but not if `blockA.Parents != blockC.Parents`.

Reply:
Ack. Could you update the diagram here (linked below) to match please?